### PR TITLE
NIAD-1548: mock mhs send ack update e2e tests

### PR DESCRIPTION
**Pull request by AlanSimpson1723**

### CODE REVIEW

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java [hunk 1]
```diff
@@ -77,8 +78,15 @@
         } else if (interactionId.equals(ACKNOWLEDGEMENT_INTERACTION_ID)) {
             LOGGER.info("Message acknowledgement accepted.");
             headers.setContentType(MediaType.TEXT_XML);
+            try{
+                var inboundMessage = STUB_ACKNOWLEDGEMENT_INBOUND_MESSAGE.replace("%%ConversationId%%", correlationId);
+                inboundProducer.sendToMhsInboundQueue(inboundMessage);
+                LOGGER.info("Message acknowledgement sent to Inbound Queue, acknowledgementInteractionId: " + ACKNOWLEDGEMENT_INTERACTION_ID);
+            }catch (JmsException e) {
+                LOGGER.error("Error could not send acknowledgement to Inbound Queue", e);
```
Comment (dawidskainos) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:86]: I think we should return an error when exception is caught. At the moment you're only logging it and then returning HTTP 202

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java [hunk 2]
```diff
@@ -77,8 +78,15 @@
...
+            }
             return new ResponseEntity<>(STUB_ACCEPTED_RESPONSE, headers, ACCEPTED);
-        } else if (interactionId.equals(COMMON_INTERACTION_ID)) {
+        }else if (interactionId.equals(COMMON_INTERACTION_ID)) {
```
Comment (dawidskainos) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:89]: [minor] code formatting - see missing space - "}else"

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 3]
```diff
@@ -45,13 +46,16 @@
     private static final String GPC_ACCESS_DOCUMENT = "gpcAccessDocument";
     private static final String EHR_EXTRACT_CORE = "ehrExtractCore";
     private static final String EHR_CONTINUE = "ehrContinue";
+    private static final String ACKNOWLEDGEMENT_INTERACTION_ID = "MCCI_IN010000UK13";
+    private static final String GPC_STRUCTURED_FILENAME_EXTENSION = "_gpc_structured.json";
```
Comment (Broom94) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:50]: These two variables dont seem to be used anywhere

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package uk.nhs.adaptors.gp2gp.e2e;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.registerFormatterForType;
```
Comment (Broom94) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:4]: This import is not used

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 5]
```diff
@@ -120,8 +124,11 @@ private void assertHappyPathWithDocs(String conversationId, String fromODSCode)
 
         waitFor(() -> assertThat(assertThatExtractCommonMessageWasSent(conversationId)).isTrue());
 
-        var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get("ackToRequester"));
+        var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(ACK_TO_REQUESTER));
         assertThatAcknowledgementToRequesterWasSent(ackToRequester, ACCEPTED_ACKNOWLEDGEMENT_TYPE_CODE);
+
+        var ackToDB = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(ACK_TO_REQUESTER));
```
Comment (Broom94) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:130]: On line 127 and 130 we make the same call to the database for this ack to requester, I would double-check this I think there another form of ack update and adjust the remain tests according

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java [hunk 6]
```diff
@@ -77,8 +78,15 @@
         } else if (interactionId.equals(ACKNOWLEDGEMENT_INTERACTION_ID)) {
             LOGGER.info("Message acknowledgement accepted.");
             headers.setContentType(MediaType.TEXT_XML);
+            try{
+                var inboundMessage = STUB_ACKNOWLEDGEMENT_INBOUND_MESSAGE.replace("%%ConversationId%%", correlationId);
+                inboundProducer.sendToMhsInboundQueue(inboundMessage);
+                LOGGER.info("Message acknowledgement sent to Inbound Queue, acknowledgementInteractionId: " + ACKNOWLEDGEMENT_INTERACTION_ID);
+            }catch (JmsException e) {
```
Comment (Broom94) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:85]: On intellij go to code -> reformat file

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java [hunk 7]
```diff
@@ -89,7 +97,6 @@
     }
 
     private void verifyOutboundMessagePayload(String requestBody) throws JsonProcessingException, MockMHSException {
-        LOGGER.debug("Received outbound MHS request payload:\n{}", requestBody);
```
Comment (Broom94) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:92]: Is there a reason why this has been removed?

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 8]
```diff
@@ -82,8 +84,8 @@ public void When_ExtractRequestReceivedForPatientWithNoDocs_Expect_DatabaseToBeU
         var gpcAccessDocument = waitFor(() -> emptyDocumentTaskIsCreated(conversationId));
         assertThatNotDocumentsWereAdded(gpcAccessDocument);
 
-        var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get("ackToRequester"));
-        assertThatAcknowledgementToRequesterWasSent(ackToRequester, ACCEPTED_ACKNOWLEDGEMENT_TYPE_CODE);
+        var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get(ACK_TO_PENDING));
```
Comment (Broom94) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:87]: Change this variable name

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java (hunk 1 above)
Comment (Broom94) [reply mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:88]: Still need to return a 202 here

File: mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java [hunk 9]
```diff
@@ -77,13 +78,21 @@
...
+            } catch (JmsException e) {
+                LOGGER.error("Error could not send acknowledgement to Inbound Queue", e);
+                return new ResponseEntity<>(INTERNAL_SERVER_ERROR_RESPONSE, headers, INTERNAL_SERVER_ERROR);
+            }
         } else if (interactionId.equals(COMMON_INTERACTION_ID)) {
             LOGGER.info("Message Common accepted.");
             headers.setContentType(MediaType.TEXT_XML);
             return new ResponseEntity<>(headers, ACCEPTED);
         }
-
```
Comment (Broom94) [inline mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java:86]: add space here

**Review by Broom94 — APPROVED** [review]

**Review by ElliotB-96 — APPROVED** [review]
LGTM

**Pull request merged**